internal/command/command.go

```diff
@@ -22,6 +22,67 @@ type Command interface {
 	Run() error                // Run runs the command
 }
 
+type CrossBuilderCommand interface {
+	RunEach(image ContainerImage) (string, string, error) // Called to build each possible architecture/OS combination
```

Comment:
Not sure about the `image` name. Should not be something like `container` ?

Reply:
I don't know. I choose image as at least it was consistently used by docker/podman/kubernetes. Otherwise kubernetes call the assemblage of image+volume on a nodes a `pod`.

Reply:
I am more confused about the `Each`... If this is iterating through architectures, where is that state passed?
I would avoid `Each` if we can be more specific, it works for a language primitive but here seems too generic.
How does this interface relate to the concrete `CrossBuilder`?

Reply:
It is iterating on an image which is hold the information regarding engine/os/architecture/release.

`CrossBuilder` is the type that process an entire os while `CrossBuilderCommand` represent each step. Maybe `CrossBuilderStep` would be better here.

Reply:
OK, I think I follow. So I'm not sure why `CrossBuilderStep` is iterating over `Image`, wouldn't the builder be responsible for iterations so the build step is just 1 step?

Reply:
Indeed, the `CrossBuilder` will iterate over all the `ContainerImage` and call `CrossBuilderStep` once to do OS specific task.

Reply:
So `RunEach` is more like just `Run`?